rpc/src/error.rs

```diff
@@ -48,12 +36,12 @@ define_error! {
             | _ | { "I/O error" },
 
         Http
-            [ HttpError ]
+            [ ReqwestError ]
             | _ | { "HTTP error" },
 
-        Hyper
-            [ HyperError ]
-            | _ | { "HTTP error" },
+        InvalidProxy
+            [ ReqwestError ]
+            | _ | { "Invalid proxy configuration" },
```

Comment:
What's the downside of exposing those? Breaking semver compatibility if we update the underlying client crate to another major version or if we change it again?

Since we are already releasing breaking changes very frequently, I don't believe that's a big deal, but I am happy to be convinced otherwise.

On the other hand, the only downside of type-erasing those errors is that whoever is consuming them will have to dynamically cast them if they want to inspect them, which is brittle.

So my vote is for keeping those as-is, but I am open to changing my mind.

Reply:
Agreed! Let's cross that bridge when we get to it